### PR TITLE
[14.0][MIG]osi_analytic_segments_defaults

### PR DESCRIPTION
**Pull request by hpatelserpentcs**
osi_analytic_segments_defaults module migrated in v14

### CODE REVIEW

File: osi_analytic_segments_defaults/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "license": "LGPL-3",
+    "summary": "Additional analytic segments for Analytic Defaults",
+    "author": "Open Source Integrators",
+    "maintainer": "Open Source Integrators",
+    "website": "http://www.opensourceintegrators.com",
+    "category": "Analytic Accounting",
+    "depends": [
+        "purchase",
+        "account",
+        "osi_analytic_segments",
```
Comment (patrickrwilson) [inline osi_analytic_segments_defaults/__manifest__.py:17]: This module also depends on sale (I received an error that sale.order.line doesn't exist in registry when trying to install it.)
Please add "sale" to depends.

File: osi_analytic_segments_defaults/views/account_analytic_default_view.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0"?>
+<odoo>
+
+    <!-- Copyright 2019 Open Source Integrators
```
Comment (patrickrwilson) [inline osi_analytic_segments_defaults/views/account_analytic_default_view.xml:4]: Update date and move to top for formatting consistancy.

**Review by patrickrwilson — CHANGES_REQUESTED** [review]
Tested and works but depends on sale and small xml change requested.

File: osi_analytic_segments_defaults/__manifest__.py (hunk 1 above)
Comment (hpatelserpentcs) [reply osi_analytic_segments_defaults/__manifest__.py:17]: Added

**Review by patrickrwilson — APPROVED** [review]
LGTM

File: osi_analytic_segments_defaults/models/account_move_line.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    # def _set_additional_fields(self, invoice):
+    #     rec = self.env['account.analytic.default'].account_get(
+    #         self.product_id.id, self.invoice_id.partner_id.id,
+    #         self.env.uid, fields.Date.today(),
+    #         company_id=invoice.company_id.id)
+    #     if rec and not self.analytic_segment_one_id:
+    #         self.analytic_segment_one_id = rec.analytic_segment_one_id.id
+    #     if rec and not self.analytic_segment_two_id:
+    #         self.analytic_segment_two_id = rec.analytic_segment_two_id.id
+    #     super(AccountMoveLine, self)._set_additional_fields(invoice)
```
Comment (smangukiya) [inline osi_analytic_segments_defaults/models/account_move_line.py]: @hpatelserpentcs I would rather remove commented code out.

**Review by smangukiya — APPROVED** [review]
Minor code clean, else looks good.

File: osi_analytic_segments_defaults/models/account_move_line.py (hunk 3 above)
Comment (hpatelserpentcs) [reply osi_analytic_segments_defaults/models/account_move_line.py]: removed

**Comment by patrickrwilson** [comment]
@hpatelserpentcs I just discovered an error, can you look into it when you have a chance. Create new invoice and try to add a line, this error shows:
Traceback (most recent call last):
  File "C:\Odoo\Odoo 14.0e\server\odoo\http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "C:\Odoo\Odoo 14.0e\server\odoo\http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
AttributeError: 'account.move.line' object has no attribute 'invoice_id'

**Comment by hpatelserpentcs** [comment]
> @hpatelserpentcs I just discovered an error, can you look into it when you have a chance. Create new invoice and try to add a line, this error shows:
> Traceback (most recent call last):
> File "C:\Odoo\Odoo 14.0e\server\odoo\http.py", line 639, in _handle_exception
> return super(JsonRequest, self)._handle_exception(exception)
> File "C:\Odoo\Odoo 14.0e\server\odoo\http.py", line 315, in _handle_exception
> raise exception.with_traceback(None) from new_cause
> AttributeError: 'account.move.line' object has no attribute 'invoice_id'

I have added the improvement in this https://github.com/ursais/osi-addons/pull/357 PR:

**Pull request merged**